### PR TITLE
Add EnableEncryptionAtHost in ManagedMachinePool

### PR DESCRIPTION
**Pull request by mihaiandreiratoiu**
**What this PR does / why we need it**:
- Add EnableEncryptionAtHost flag in ManagedMachinePool 

**TODOs**:

- [X] squashed commits
- [X] includes documentation
- [X] adds unit tests

**Release note**:

```release-note
 - Add EnableEncryptionAtHost in ManagedMachinePool
```

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
@mihaiandreiratoiu please take a look at failing tests

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/assign @nojnhuh

**Comment by dtzar** [comment]
@nojnhuh - looks like codecov just failed to upload results, so just needs an approval to merge?

**Comment by CecileRobertMichon** [comment]
@dtzar I just triggered codecov

I assigned @nojnhuh to this one because it conflicts with #3529 so want to make sure he has a chance to review/integrate with his PR

**Comment by nojnhuh** [comment]
At first glance I don't see anything that conflicts with #4069 more than any other feature, so I'm pretty confident this won't need changes to accommodate ASO but I'll review this when I get the chance.

File: api/v1beta1/azuremanagedmachinepool_types.go [hunk 1]
```diff
@@ -561,6 +561,11 @@ type AzureManagedMachinePoolSpec struct {
 	// Immutable.
 	// +optional
 	EnableFIPS *bool `json:"enableFIPS,omitempty"`
+
+	// EnableEncryptionAtHost indicates whether host encryption is enabled on the node pool.
+	// Immutable.
```
Comment (nojnhuh) [inline api/v1beta1/azuremanagedmachinepool_types.go:566]: Could we also include a link to the AKS docs for this feature like this?
```suggestion
	// Immutable.
	// See also [AKS doc].
	//
	// [AKS doc]: https://learn.microsoft.com/en-us/azure/aks/enable-host-encryption
```

**Review by nojnhuh — COMMENTED** [review]
/lgtm
/approve

Thank you @mihaiandreiratoiu!

**Pull request merged**